File: src/frontend/src/services/evaluationService.ts
import api from './api';

export interface SurveyDto {
  surveyId: number;
  title: string;
  description: string;
  openDate: string;
  closeDate: string;
  status: 'Pending' | 'Completed' | 'Closed' | 'Draft';
  submittedAnswers?: string;
  submittedAt?: string;
}

export interface SurveyResponseRequest {
  responses: Record<string, any>;
  comments: string;
}

export const evaluationService = {
  getSurveys: async (): Promise<SurveyDto[]> => {
    return await api.get<SurveyDto[]>('/api/v1/evaluations');
  },

  getSurveyById: async (id: number): Promise<SurveyDto> => {
    return await api.get<SurveyDto>(`/api/v1/evaluations/${id}`);
  },

  submitSurvey: async (id: number, data: SurveyResponseRequest): Promise<SurveyDto> => {
    return await api.post<SurveyDto>(`/api/v1/evaluations/${id}`, data);
  },
};

File: src/backend/src/main/java/com/myus/service/EvaluationService.java
package com.myus.service;

import com.myus.dto.SurveyDto;
import com.myus.dto.SurveyResponseRequest;

import java.util.List;

public interface EvaluationService {
    List<SurveyDto> getSurveysForStudent(String username);
    SurveyDto getSurveyById(String username, Long surveyId);
    SurveyDto submitSurvey(String username, Long surveyId, SurveyResponseRequest request);
}
